observability/probe/aggregate.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package probe
+
+import (
+	"github.com/juju/errors"
+)
+
+type Aggregate struct {
```

Comment:
should this be exported?
we have the NewAggregate function below (which needs a comment)